### PR TITLE
allow changing server command line from http->https

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
allow changing server command line from http->https

## Motivation and Context
this is allowed as long as order is preserved as is
on an existing setup, the new command line is updated
in `pool.bin` to facilitate future decommission's on
these pools.

## How to test this PR?

You may use the following YAML to start the service and make sure to generate certs for `localhost,minio1,minio2,minio3,minio4` using https://github.com/minio/certgen

```yaml
version: '3.7'

# Settings and configurations that are common for all containers
x-minio-common: &minio-common
  image: y4m4/minio:dev
  command: server --console-address ":9001" http://minio{1...4}/data{1...2} http://minio{1...4}/data{3...4} http://minio{1...4}/data{5...6}
  # command: server --console-address ":9001" https://minio{1...4}/data{1...2} https://minio{1...4}/data{3...4} https://minio{1...4}/data{5...6}
  expose:
    - "9000"
    - "9001"
  environment:
    _MINIO_SERVER_DEBUG: "on"
  healthcheck:
    test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
    interval: 30s
    timeout: 20s
    retries: 3

# starts 4 docker containers running minio server instances.
# using nginx reverse proxy, load balancing, you can access
# it through port 9000.
services:
  minio1:
    <<: *minio-common
    hostname: minio1
    volumes:
      - /home/harsha/.minio/:/root/.minio
      - data1-1:/data1
      - data2-1:/data2
      - data3-1:/data3
      - data4-1:/data4
      - data5-1:/data5
      - data6-1:/data6

  minio2:
    <<: *minio-common
    hostname: minio2
    volumes:
      - /home/harsha/.minio/:/root/.minio
      - data1-2:/data1
      - data2-2:/data2
      - data3-2:/data3
      - data4-2:/data4
      - data5-2:/data5
      - data6-2:/data6

  minio3:
    <<: *minio-common
    hostname: minio3
    volumes:
      - /home/harsha/.minio/:/root/.minio
      - data1-3:/data1
      - data2-3:/data2
      - data3-3:/data3
      - data4-3:/data4
      - data5-3:/data5
      - data6-3:/data6

  minio4:
    <<: *minio-common
    hostname: minio4
    volumes:
      - /home/harsha/.minio/:/root/.minio
      - data1-4:/data1
      - data2-4:/data2
      - data3-4:/data3
      - data4-4:/data4
      - data5-4:/data5
      - data6-4:/data6

  nginx:
    image: nginx:1.19.2-alpine
    hostname: nginx
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf:ro
    ports:
      - "9000:9000"
      - "9001:9001"
    depends_on:
      - minio1
      - minio2
      - minio3
      - minio4

## By default this config uses default local driver,
## For custom volumes replace with volume driver configuration.
volumes:
  data1-1:
  data1-2:
  data1-3:
  data1-4:
  data2-1:
  data2-2:
  data2-3:
  data2-4:
  data3-1:
  data3-2:
  data3-3:
  data3-4:
  data4-1:
  data4-2:
  data4-3:
  data4-4:
  data5-1:
  data5-2:
  data5-3:
  data5-4:
  data6-1:
  data6-2:
  data6-3:
  data6-4:
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-server-pool-decom.go [hunk 1]
```diff
@@ -323,6 +346,12 @@ func (p *poolMeta) validate(pools []*erasureSets) (bool, error) {
 			continue
 		}
 		_, ok := specifiedPools[k]
+		if !ok {
```
Comment (krisis) [inline cmd/erasure-server-pool-decom.go:349]: IIUC, we need to update the url scheme only for endpoint URLs in `rememberedPool` loaded from `pool.bin` on startup. The endpoint URLs in `specifiedPools` are based on the latest command line, so they shouldn't require any modification.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-server-pool-decom.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/erasure-server-pool-decom.go:349]: the reason its done so to not modify what is on the disk right way, just figure out what is causing the change and verifying if the change is correct from a command line point of view. 

this is why we verify is the "Remembered" part correct even with the changes from the user.

**Review by krisis — APPROVED** [review]

**Review by ravindk89 — APPROVED** [review]
This looks fine to my limited knowledge of how the systems work.

**Pull request merged**